assets/js/collaborative-editor/components/LoadingBoundary.tsx

```diff
@@ -38,6 +38,11 @@ export function LoadingBoundary({ children }: LoadingBoundaryProps) {
     hasWorkflowData &&
     (session.settled || !isInitialLoad);
 
+  // most of the states we depend on don't need to show the loading workflow screen I guess.
+  const subReady = hasWorkflowData && !isInitialLoad;
+
+  if (subReady) return <>{children}</>;
+
```

Comment:
Initially, the loading screen is shown when
1. session context loading state changes
2. we don't have workflow data yet (**needed**)
3. session has settled `session.settled`
4. it's our initial load (**needed**)

Those marked **needed** are the ones this PR gives priority to and agrees that we don't show loading state if they've not changed.